test/pthread/test_pthread_key_recreation.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <iostream>
+#include <cassert>
+#include <condition_variable>
+#include <pthread.h>
+#include <emscripten.h>
+
+// tests, in the following sequence
+//  pthread_key_create with key k
+//  pthread_key_delete of key k
+//  pthread_key_create returning the same key k
+// , that the thread local slot is correctly initialized to 0 in all threads
+
+// overall structure based on test_pthread_reltime.cpp
+
+static pthread_key_t key;
+
+static std::mutex mutex;
+static std::condition_variable cond_var;
```

Comment:
It seems a little odd to make pthread-API stuff with C++ threed API stuff, but maybe its fine?

The alternative would be to use `pthread_mutex_t` and `pthread_cond_t` here (you could also then make this in a pure C test without any C++).